api-reference/_hidden/dxDataGridColumn/calculateGroupValue.md

```diff
@@ -13,9 +13,311 @@ The current row's data.
+    </dx-data-grid>
+
+    <!-- tab: app.module.ts -->
+    import { BrowserModule } from '@angular/platform-browser'; 
+    import { NgModule } from '@angular/core'; 
+    import { AppComponent } from './app.component'; 
+    import { DxDataGridModule } from 'devextreme-angular'; 
+    
+    @NgModule({ 
+        declarations: [ 
+            AppComponent 
+        ], 
+        imports: [ 
+            BrowserModule, 
+            DxDataGridModule 
+        ], 
+        providers: [ ], 
+        bootstrap: [AppComponent] 
+    }) 
+    
+    export class AppModule { }
+    
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxDataGrid ... >
+            <DxColumn
+                data-field="ColumnValues"
+                :group-index="0"
+                calculate-group-value="GroupingValues" 
+            />
+        </DxDataGrid>
+    </template>
+    <script>
+    import { DxDataGrid, DxColumn } from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid,
+            DxColumn
+        },
+        // ...
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+    import DataGrid, { Column } from 'devextreme-react/data-grid';
+
+    function App() {
+        // ...
+        return (
+            <DataGrid ...>
+                <Column
+                    dataField="ColumnValues"
+                    groupIndex={0}
+                    calculateGroupValue="GroupingValues" 
+                />
+            </DataGrid>
+        );
+    }
+
+    export default App();
+
+---
+
+... or a function that creates custom groups. In the code below, **calculateGroupValue** creates groups like Today, Yesterday, and Earlier:
+
+
+---
+##### jQuery
+
+    <!--JavaScript-->
+    const getDayDifference = (date1, date2) => {
+        let timeDiff =Math.abs(date1.getTime() - date2.getTime());
+        return Math.ceil(timeDiff / (1000 * 3600 * 24));  
+    }
+
+    $(function() {
+        $("#dataGridContainer").dxDataGrid({
+            columns: [
+                // ...
+                {
+                    dataField: "HireDate",
+                    dataType: "date",
+                    groupIndex: 0,
+                    calculateGroupValue: function(rowData) {       
+                        const currentHireDate = rowData.HireDate;
+                        const today = new Date();
+                        today.setHours(0,0,0,0);
+
+                        const dayDifference = getDayDifference(today, currentHireDate);
+                        switch (dayDifference) {
+                            case 0:
+                            return "Today";
+                            break;
+                        case 1:
+                            return "Yesterday";
+                            break; 
+                            // ...
+                        default:
+                            return "Earlier";
+                        };    
```

Comment:
If you use `return`, you don't need a `break`. Also, we can make the statement more compact.

```suggestion
                        switch (dayDifference) {
                            case 0: return "Today";
                            case 1: return "Yesterday";
                            // ...
                            default: return "Earlier";
                        }
```